src/authenticationinapp/authenticationinapp.cpp

```diff
@@ -44,13 +44,30 @@ AuthenticationInApp::~AuthenticationInApp() {
   s_instance = nullptr;
 }
 
-void AuthenticationInApp::setState(State state) {
+void AuthenticationInApp::setState(State state,
+                                   AuthenticationInAppSession* session) {
   m_state = state;
   emit stateChanged();
 
-  emit MozillaVPN::instance()->recordGleanEventWithExtraKeys(
-      GleanSample::authenticationInappStep,
-      {{"state", QVariant::fromValue(state).toString()}});
+  if (session && session->type() == AuthenticationInAppSession::TypeDefault) {
```

Comment:
what about a `switch` ?
if (session) {
  char* gleanSample = nullptr;
  switch (session->type()) {
  case  AuthenticationInAppSession::TypeDefault: gleanSample = GleanSample::authenticationInappStep; break;
  ...
  default:
   logger.error() << "Glean samples and Auth-in-app session types are out of sync";
    }

 Q_ASSERT(gleanSample);
emit MozillaVPN::instance()->recordGleanEventWithExtraKeys(gleanSample, {{"state", QVariant::fromValue(state).toString()}});
```